ui/src/app/views/copy-notebook-modal/component.tsx

```diff
@@ -0,0 +1,185 @@
+import * as React from 'react';
+
+import { Button } from 'app/components/buttons';
+import { styles as headerStyles } from 'app/components/headers';
+import { Select, TextInput } from 'app/components/inputs';
+import { Modal, ModalBody, ModalFooter, ModalTitle } from 'app/components/modals';
+import { FileDetail, Workspace } from 'generated/fetch';
+
+import { Spinner } from 'app/components/spinners';
+import { workspacesApi } from 'app/services/swagger-fetch-clients';
+import { WorkspacePermissions } from 'app/utils/workspace-permissions';
+
+enum State { FORM, ERROR, SUCCESS }
+
+export interface CopyNotebookModalProps {
+  fromWorkspaceNamespace: string;
+  fromWorkspaceName: string;
+  fromNotebook: FileDetail;
+  onClose: Function;
+  onCopy: Function;
+}
+
+export interface CopyNotebookModalState {
+  writeableWorkspaces: Array<Workspace>;
+  destination: Workspace;
+  newName: string;
+  state: State;
+  errorMsg: string;
+  loading: boolean;
+}
+
+const boldStyle = {
+  fontWeight: 700
+};
+
+export class CopyNotebookModal extends React.Component<CopyNotebookModalProps,
+CopyNotebookModalState> {
+  constructor(props: CopyNotebookModalProps) {
+    super(props);
+
+    this.state = {
+      writeableWorkspaces: [],
+      newName: props.fromNotebook.name,
+      destination: null,
+      state: State.FORM,
```

Comment:
Nit: I would name this something about what state its talking about. Is this `copyNotebookResponseState`? In general I prefer verboseness when it gives more clarity.